File: target-dotnet/src/SESS.NexaERP.Domain/Masters/Vendor.cs
using SESS.NexaERP.Domain.Common;

namespace SESS.NexaERP.Domain.Masters;

public sealed class Vendor : AuditableEntity
{
    public string VendorCode { get; set; } = string.Empty;
    public bool IsVendorCodeLocked { get; set; }
    public string Name { get; set; } = string.Empty;
    public string LegalVendorName { get; set; } = string.Empty;
    public string? TradeName { get; set; }
    public string VendorType { get; set; } = string.Empty;
    public string? GstNumber { get; set; }
    public string? PanNumber { get; set; }
    public bool MsmeStatus { get; set; }
    public string? MsmeNumber { get; set; }
    public string? ContactPerson { get; set; }
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? BillingAddress { get; set; }
    public string? ShippingAddress { get; set; }
    public string? State { get; set; }
    public string? StateCode { get; set; }
    public string Country { get; set; } = "India";
    public string? MaterialServiceCategories { get; set; }
    public string? ApprovedMakes { get; set; }
    public string? PaymentTerms { get; set; }
    public string? DeliveryTerms { get; set; }
    public int? CreditPeriodDays { get; set; }
    public string? BankMetadataJson { get; set; }
    public string? AttachmentMetadataJson { get; set; }
    public string PortalOrganizationId { get; set; } = string.Empty;
    public string ApprovalStatus { get; set; } = MasterApprovalStatuses.Draft;
    public string VendorStatus { get; set; } = MasterStatuses.Draft;
    public string CommercialVerificationStatus { get; set; } = MasterApprovalStatuses.Draft;
    public string? CommercialVerifiedBy { get; set; }
    public DateTimeOffset? CommercialVerifiedAt { get; set; }
    public DateOnly EffectiveFrom { get; set; } = DateOnly.FromDateTime(DateTime.UtcNow);
    public DateOnly? EffectiveTo { get; set; }
    public bool RequiresReverification { get; set; }
    public string? ApprovedBy { get; set; }
    public DateTimeOffset? ApprovedAt { get; set; }
    public bool IsActive { get; set; } = true;
}

File: target-dotnet/src/SESS.NexaERP.Web/src/types/vendor.ts
// Mirrors SESS.NexaERP.Application.Masters vendor contracts (PascalCase wire contract).

export interface VendorSummary {
  Id: string
  VendorCode: string
  Name: string
  GstNumber: string | null
  PanNumber: string | null
  ApprovalStatus: string
  VendorStatus: string
  IsActive: boolean
  Version: number
  BankMetadata: unknown | null
}

export interface VendorDetail {
  Id: string
  VendorCode: string
  Name: string
  LegalVendorName: string
  TradeName: string | null
  VendorType: string
  GstNumber: string | null
  PanNumber: string | null
  MsmeStatus: boolean
  MsmeNumber: string | null
  ContactPerson: string | null
  Phone: string | null
  Email: string | null
  BillingAddress: string | null
  ShippingAddress: string | null
  State: string | null
  StateCode: string | null
  Country: string
  MaterialServiceCategories: string | null
  ApprovedMakes: string | null
  PaymentTerms: string | null
  DeliveryTerms: string | null
  CreditPeriodDays: number | null
  BankMetadata: unknown | null
  AttachmentMetadataJson: string | null
  ApprovalStatus: string
  VendorStatus: string
  IsActive: boolean
  Version: number
}

export interface UpsertVendorRequest {
  VendorCode: string
  LegalVendorName: string
  TradeName: string | null
  VendorType: string
  GstNumber: string | null
  PanNumber: string | null
  MsmeStatus: boolean
  MsmeNumber: string | null
  ContactPerson: string | null
  Phone: string | null
  Email: string | null
  BillingAddress: string | null
  ShippingAddress: string | null
  State: string | null
  StateCode: string | null
  Country: string
  MaterialServiceCategories: string | null
  ApprovedMakes: string | null
  PaymentTerms: string | null
  DeliveryTerms: string | null
  CreditPeriodDays: number | null
  BankMetadataJson: string | null
  AttachmentMetadataJson: string | null
  Version: number | null
}
